### ISSUE TITLE
Questions about the DELG model

### ISSUE THREAD

**Issue by XieGang729** [tensorflow/models#9438]
1. The entire URL of the documentation with the issue

https://github.com/tensorflow/models/tree/master/research/delf/delf/python/delg

2. Describe the issue
hi！@andrefaraujo ，How is it going？
1.According to the paper _'Unifying Deep Local and Global Features for Image Search'_,In the process of extracting global features, does the model only use the GeM pooling technology?The effect is great when I used delg_gld_20200520 in my datasets,it's amazing!
2.If I want to train a model for extracting global features as shown in the paper, can you give me some help from the labeling of the data set and the setting of the network model?
3.Having said that, the process of image retrieval is to extract features and calculate distances. Why do we need image annotation? Is it because the model that extracts features needs to be trained through annotated data set? If so, the model for extracting features is more similar to a classifier, isn't it?
I am a beginner in computer vision, please don’t mind if my question seems relatively low-level，thank you,andre!

**Comment by XieGang729** [comment]
@andrefaraujo Would you like to give me some help at your convenience?:D

**Comment by andrefaraujo** [comment]
thanks for your interest! :)

1) Yes, it's using GeM with multi-scale extraction (image pyramid).
2) You only would need image-level labels, like the ones we have for the landmarks dataset. You could follow the instructions for the landmarks dataset training [here](https://github.com/tensorflow/models/blob/master/research/delf/delf/python/training/README.md), then modify to use your own dataset. If you are only interested in the global features, you could set `attention_loss_weight ` to 0 when running the training, which will make it ignore the local feature part.
3) Yes, we use image annotations to train the model. We train the model with a proxy classification loss, but at inference time we discard the classifier and keep only the features.
